packages/components/src/truncate/utils.js

```diff
@@ -2,6 +2,8 @@
  * External dependencies
  */
 import { isNil } from 'lodash';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import stringWidth from 'string-width';
```

Comment:
Thanks for persisting with this one. 

Things work as described in the browser, and functionality-wise, a good addition to the Truncate component I think.

<img width="279" alt="Screen Shot 2022-06-02 at 2 28 12 pm" src="https://user-images.githubusercontent.com/6458278/171553022-5fb2046a-b7fe-4150-a32b-de0d058d2aa5.png">

My only question would be to ask whether the library is appropriate. 

This package and its dependencies aren't actively maintained, and I'm wondering the utility is worth the bump in package size:

```bash
| `build/components/index.min.js` | 233 kB | +5.01 kB (+2%) |
```

(from the [CI reports](https://github.com/WordPress/gutenberg/runs/6670072210?check_suite_focus=true#step:4:934))

Plus jest doesn't seem to be able to compile the library via Babel.

Even so, do we care about possible ANSI or emoji characters?

Taking those away, it seems as if `string-width` [relies mainly on eastasianwidth](https://github.com/sindresorhus/string-width/blob/main/index.js#L39)

Maybe we could get away with just using https://github.com/komagata/eastasianwidth?

I'm not sure. What do you think?